bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/values/XmlSequence.java

```diff
@@ -675,6 +675,14 @@ public boolean equals(Object o, Set<ValuePair> visitedValues) {
             return this.getChildrenList().size() == 1 &&
                     isEqual(this.getChildrenList().get(0), o);
         }
+        if (o instanceof XmlText rhsXMLText) {
```

Comment:
This logic will break if rhdType is xml never and child list is empty.

Reply:
I'll fix this.

Reply:
The logic was changed due to other changes related to `xmlSequence`. Refer to eee266780c1186097bc83dcc8f5ab3cf30ae5aee and later commits for updates.